src/redis-pub-sub.ts

```diff
@@ -0,0 +1,42 @@
+import type redis from 'redis';
+import type { GenericPubSub } from 'boardgame.io/server';
+
+export class RedisPubSub<T> implements GenericPubSub<T> {
+  private pubClient: redis.RedisClient;
+  private subClient: redis.RedisClient;
+  callbacks: Map<string, ((payload: T) => void)[]> = new Map();
+
+  constructor(pubClient: redis.RedisClient, subClient: redis.RedisClient) {
```

Comment:
You’re going to be the main consumer of this repo to start with, so if you prefer to keep this as is, that’s fine, but my suggestion was to allow both simple and complex set-ups:

```suggestion
  constructor({
    clientConfig,
    pubClientConfig = clientConfig,
    subClientConfig = clientConfig,
    pubClient = redis.createClient(pubClientConfig),
    subClient = redis.createClient(subClientConfig),
  }) {
    this.pubClient = pubClient;
    this.subClient = subClient;
  }
```

#### Usage examples

```js
new RedisPubSub({
  clientConfig: { host: '192.168.8.50' },
});
```

```js
new RedisPubSub({
  pubClientConfig: { host: '192.168.8.50' },
  subClientConfig: { host: '192.168.8.50' }
});
```

```js
new RedisPubSub({
  pubClient: redis.createClient({ host: '192.168.8.50' }),
  subClient: redis.createClient({ host: '192.168.8.50' }),
});
```

Reply:
I dont want to add unneeded complexity for now... Requiring two redis client seems to also be the pattern on other  very popular libraries: https://socket.io/docs/v4/redis-adapter/

Reply:
Sounds good.